### ISSUE TITLE
Allow no-dupe-scenario-names to only flag duplicate scenarios inside one feature

### ISSUE THREAD

**Issue by nene** [vsiakka/gherkin-lint#64]
The current rule looks for scenario name duplicates globally. For my purposes I don't find this useful as I have several features where I want to test the same scenario. But I can imagine some might want all their scenarios to be uniquely named, and it's fine.

However if there are duplicate scenario names within the same feature, then that's clearly an error and I'd like gherkin-lint to catch this.

I'd propose configuration syntax like:

```js
"no-dupe-scenario-names": ["on", "global"], // the default (current behavior)
"no-dupe-scenario-names": ["on", "feature-local"],
```

Or perhaps:

```js
"no-dupe-scenario-names": ["on", "anywhere"], // the default (current behavior)
"no-dupe-scenario-names": ["on", "in-feature"],
```

Not really sure what the good names for these config options should be.

An alternative would be to have a completely separate rule, but that wouldn't make sense as one would never combine the two rules.

I'm willing to do a PR for this, but I'd like to settle the naming question first.

**Comment by josemenveg** [comment]
I'm working in this issue, and I have found a problem with the unit test.

If I've understood the code; when testing, we can launch the rule against a single file, and with this rule, it's needed to lauch it against several files (to check what happens when the scenario name is in several files).

I'm going to solve this question in the same pull.

**Comment by josemenveg** [comment]
Ok, I've a first version of this change in my [fork](https://github.com/josemenveg/gherkin-lint).

To configure it, it's a bit different.
` "no-dupe-scenario-names": ["on", {"local": "yes"}]`

I wanted to solve  issue #94 before making the pull request.

I'm opent to suggestions.

**Comment by josemenveg** [comment]
I have a pull for this issue, check if it works as expected

**Comment by xgbuils** [comment]
Hi @nene, I think that `anywhere` and `in-feature` are the two more expressive and concise options for me.

**Comment by vsiakka** [comment]
Change released with v3.2.0
https://github.com/vsiakka/gherkin-lint#no-dupe-scenario-names
